### ISSUE TITLE
Cargo fix: Could not fix use statements

### ISSUE THREAD

**Issue by KillTheMule** [rust-lang/cargo#5767]
Hey!

I've tried applying cargo fix to transition my crate to edition 2018, following the official guide. The manual preparations would be in [this commit](https://github.com/KillTheMule/nvimpam/commit/6ecb87fbf085b6a860ccaab06ba540453a5c343b). After that, the only thing to do for me was [fix use statements](https://github.com/KillTheMule/nvimpam/commit/223938f4480ff5d8dbc6c2d35223f9f3a24f1dbc). Nothing special there, all works as advertised. Except: Cargo fix was not able to automatically fix the use statements. I'm not sure [if it's supposed to](https://www.reddit.com/r/rust/comments/90kt6h/2018_edition_end_of_week_post_20180720/e2s3g1x/), but I went with "When in doubt, issue".

An additional thing is that it wasn't exactly clear to me what was going on. From its output, it looked like cargo fix tried to compile my crate, which obviously failed with the expected errors. But nothing gave an indication if cargo fix encountered a problem (e.g. something it couldn't fix), or whatever went wrong (besides the non-correct use-statements, of course).

If I should provide more information, just let me know what you need.

**Comment by alexcrichton** [comment]
Thanks for the report @KillTheMule! Always better to err on the side of opening an issue :)

I think the problem here was that the 2018 edition was enabled too soon. [The edition guide](https://rust-lang-nursery.github.io/edition-guide/editions/transitioning.html) has some detailed steps and it's currently important to *not* enable the edition before transitioning to the 2018 edition. That allows `cargo fix` to handle the warnings that would otherwise be compiler errors (before they are errors).

Does that fix the issue for you?

**Comment by KillTheMule** [comment]
It's somewhat unclear what steps I should take, probably because there's no obvious (to me...) difference between "enabling the edition" and "transitioning to the edition".

I just went back to [this commit](https://github.com/KillTheMule/nvimpam/commit/c9d870235b3cd1a7de53cceadb641b4b0372b5eb), that is, the one I started with anyways, and tried to run `cargo +nightly fix --prepare-for 2018 --all-targets --all-features` from the guide. Results in `error: Found argument '--all-targets' which wasn't expected, or isn't valid in this context`. Removed it, then got the analogous error about `--all-features`.

Then it went... weird. Ran through, seemed to be checking (not sure what that mean, probably as in `cargo check`) everything, finished without error... but didn't do anything else, neither suggest changes nor do them. Then I put the feature flag in the lib and binary files... and now it worked!

The guide seems pretty clear, although I'd add a sentence of warning. But maybe I'm the only one who's confused. The confusion probably stems from nrc's blog post, which mentions cargo fix, but (at least for me) left the exact meaning of "transitioning" unclear.

Thanks!
